src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -771,6 +751,59 @@ static int add_root_mount(void) {
 #endif
 }
 
+static int process_loader_partitions(DissectedPartition *esp, DissectedPartition *xbootldr) {
+        sd_id128_t loader_uuid;
+        int r, ret = 0;
+
+        assert(esp);
+        assert(xbootldr);
+
+        if (!is_efi_boot()) {
+                log_debug("Not an EFI boot, skipping loader partition UUID check.");
+                goto mount;
+        }
+
+        /* Let's check if LoaderDevicePartUUID points to either ESP or XBOOTLDR. We prefer it pointing
+         * to the ESP, but we accept XBOOTLDR too. If it points to neither of them, don't mount any
+         * loader partitions, since they are not the ones used for booting. */
+
+        r = efi_loader_get_device_part_uuid(&loader_uuid);
+        if (r < 0) {
+                if (r == -ENOENT) {
+                        log_debug_errno(r, "EFI loader partition unknown, skipping ESP and XBOOTLDR mounts.");
+                        return 0;
+                }
+
+                return log_debug_errno(r, "Failed to read loader partition UUID, ignoring: %m");
+        }
+
+        if (esp->found && sd_id128_equal(esp->uuid, loader_uuid))
+                goto mount;
+
+        if (xbootldr->found && sd_id128_equal(xbootldr->uuid, loader_uuid)) {
+                log_debug("LoaderDevicePartUUID points to XBOOTLDR partition, proceeding anyway.");
```

Comment:
"anyway" suggests this was a problem of some sort. I think the message generally makes sense for informational reasons, but I'd drop the ", proceeding anyway".